src/herring/tl_helpers.hpp

```diff
@@ -151,10 +153,36 @@ auto convert_tree(treelite::Tree<tl_threshold_t, tl_output_t> const& tl_tree, bo
           throw unconvertible_model_exception{"Unsupported comparison operator"};
         }
       } else {
-        throw unconvertible_model_exception{"Categorical nodes not yet implemented"};
+        if (tl_tree.CategoriesListRightChild(cur_node_id)) {
+          hot_child = left_id;
+          distant_child = right_id;
+        } else {
+          hot_child = right_id;
+          distant_child = left_id;
+        }
+        auto tl_categories = tl_tree.MatchingCategories(cur_node_id);
+        auto constexpr max_category = typename tree_t::node_type::category_set_type{}.size();
+        auto bit_representation = std::transform_reduce(
+          std::begin(tl_categories),
+          std::end(tl_categories),
+          std::size_t{},
```

Comment:
Could it be possible for `1 << max_category` to be too large to fit in `std::size_t`?